### ISSUE TITLE
dartdoc version shipped with 3.2.0 has broken sidebar

### ISSUE THREAD

**Issue by parlough** [dart-lang/sdk#54067]
The left sidenav is missing on generated documentation for non-class templates/pages.

Example: https://api.dart.dev/stable/3.2.0/dart-ffi/NativeCallable/NativeCallable.isolateLocal.html

On the stable branch (3.2.0) dartdoc is at https://github.com/dart-lang/dartdoc/commit/a3cfdc40a94d169bd7b559dc6f83ceb7f79de4e2. 

There were a few related dartdoc fixes that landed after the beta cutoff:

- https://github.com/dart-lang/dartdoc/commit/c147aa19a2a655a2cf9af7a8e6a0f104e6116137
  - This caused another issue which was fixed by https://github.com/dart-lang/dartdoc/commit/4283dbc60f40539d7d8aca4b29e6e14b7607dd0d
- https://github.com/dart-lang/dartdoc/commit/5156398c17ff7ab8defd87c7a5b9de1ee7e4564a

dartdoc [v7.0.1](https://pub.dev/packages/dartdoc/changelog#701) should include all of these fixes, but it is a relatively [wide range of commits](https://github.com/dart-lang/dartdoc/compare/a3cfdc40a94d169bd7b559dc6f83ceb7f79de4e2...b350c68850f0662624dd870a720e97e18f122eee), so I'm not sure if that's acceptable from a cherrypick perspective?

**Comment by bkonyi** [comment]
FYI @srawlins

**Comment by srawlins** [comment]
Ah, but this is of course a _cherry-pick_ of a commit sitting in SDK main branch. So which commit do we pick? The DEPS rolls often include more than one package which would make it a much spookier cherry-pick, but luckily the commit that introduces the 3rd fix to the sidebars is https://github.com/dart-lang/sdk/commit/279f6d77aed890abfdd0d93341d8b94239dca610, which includes only one more local commit (https://github.com/dart-lang/dartdoc/commit/8c103397734f4dba75915e40c734a88e29a35f26), a harmless docs change.

**Comment by parlough** [comment]
Thanks for your thoughts and insight @srawlins. I agree that's a good commit to use and nothing in that range seems concerning. The original range included canonicalization changes, so this seems much better. 

I opened https://github.com/dart-lang/sdk/issues/54073 and https://dart-review.googlesource.com/c/sdk/+/336780, but I've never created a cherry-pick request before, so let me know if I made any mistakes. Thank you!
